### DISCUSSION TITLE
math.sin(x) can not return correct result when x is ralatively big number

### DISCUSSION THREAD

**Discussion by flamehe0580** [vlang/v#17342] [Questions and Answers]
>>> x:=[1.0, 1e5, 1e10, 1e15, 1e20, 1e30, 1e40]
>>> x
[1.0, 100000.0, 1.e+10, 1.e+15, 1.e+20, 1.e+30, 1.e+40]
>>> for a in x {
... println("sin($a) = ${math.sin(a)}")
... }
sin(1.0) = 0.8414709848078965        
sin(100000.0) = 0.03574879797201651  
sin(1.e+10) = -0.4875068577590483    
sin(1.e+15) = 0.8776128590254704     
sin(1.e+20) = 9.155440021830429e+41  
sin(1.e+30) = 2.5204768077725173e+182
sin(1.e+40) = -inf

obviously, it can be observed that sin(1e20) can not be greater than 1. it is a bug of function **math.sin**.

**Comment by Wajinn** [comment]
Have you tried V's VSL (V Scientific Library)?  Since you working with very big numbers, you might want to look there too.
https://vlang.github.io/vsl/

**Reply by flamehe0580** [reply]
i tried the sin of vsl.fun, the same problem

**Comment by j-bm** [comment]
Using sin(x) where x is a very large multiple of pi results in imprecise results.  Recall f64 has only 56 bits of precision.  Taking math.sin() of any number bigger than pow(2,56) * math.pi is going to be rounded down to the same number.

In fact the V implementation of sin() works in part on the integer part of x/8; if this is not expressible as an f64 (it is more than pow(2.56)) then the result is going to be incorrect, irrespective of whether it produces an error.

In practical terms nobody does this.  Do you have a particular application that needs this?

**Reply by flamehe0580** [reply]
As a programming language, i think, V should support all the floating point numbers of type f64. 
I have checked some other languages: Julia, fortran, node.js, python, MATLAB, finding that all of them support function sin to compute big numbers.
